src/algorithms/acquisition/adapters/gps_l1_ca_pcps_quicksync_acquisition.cc

```diff
@@ -19,250 +19,70 @@
 #include "gps_l1_ca_pcps_quicksync_acquisition.h"
 #include "GPS_L1_CA.h"
 #include "configuration_interface.h"
-#include "gnss_sdr_flags.h"
 #include "gps_sdr_signal_replica.h"
+#include "pcps_quicksync_acquisition_cc.h"
 #include <boost/math/distributions/exponential.hpp>
-#include <algorithm>
 
 #if USE_GLOG_AND_GFLAGS
 #include <glog/logging.h>
 #else
 #include <absl/log/log.h>
 #endif
 
-#if HAS_STD_SPAN
-#include <span>
-namespace own = std;
-#else
-#include <gsl-lite/gsl-lite.hpp>
-namespace own = gsl_lite;
-#endif
+namespace
+{
+uint32_t get_folding_factor(const ConfigurationInterface* configuration, const std::string& role)
+{
+    const int64_t fs_in_deprecated = configuration->property("GNSS-SDR.internal_fs_hz", static_cast<int64_t>(4000000));
+    const auto fs_in = configuration->property("GNSS-SDR.internal_fs_sps", fs_in_deprecated);
+    const auto code_length = static_cast<unsigned int>(round(fs_in / (GPS_L1_CA_CODE_RATE_CPS / GPS_L1_CA_CODE_LENGTH_CHIPS)));
+    const auto folding_factor = static_cast<unsigned int>(ceil(sqrt(log2(code_length))));
+    return configuration->property(role + ".folding_factor", folding_factor);
+}
```

Comment:
Having to call this twice and read the config multiple time for the same properties isn't ideal, but I think is good enough for now